src/vpn-api/utils/openvpn.ts

```diff
@@ -0,0 +1,318 @@
+/*
+	Copyright (C) 2019 Balena Ltd.
+
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published
+	by the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import * as Bluebird from 'bluebird';
+import { ChildProcess, spawn } from 'child_process';
+import { EventEmitter } from 'events';
+import * as net from 'net';
+import VpnConnector = require('telnet-openvpn');
+
+import { Netmask } from './netmask';
+
+const parsePossibleInt = (s: string): number | string => {
+	if (/^[1-9]\d*$/.test(s)) {
+		return parseInt(s, 10);
+	}
+	return s;
+};
+
+export interface VpnClientUntrustedData {
+	username: string;
+}
+export interface VpnClientTrustedData {
+	common_name: string;
+	ifconfig_pool_remote_ip: string;
+	trusted_ip: string;
+	trusted_port: number;
+}
+export interface VpnClientBytecountData {
+	bytes_received: number;
+	bytes_sent: number;
+}
+// CLIENT:CONNECT
+export interface VpnClientConnectData extends VpnClientUntrustedData {
+	password: string;
+}
+// CLIENT:ESTABLISHED
+export interface VpnClientEstablishedData
+	extends VpnClientUntrustedData,
+		VpnClientTrustedData {}
+// CLIENT:DISCONNECT
+export interface VpnClientDisconnectData
+	extends VpnClientUntrustedData,
+		Partial<VpnClientTrustedData>,
+		Partial<VpnClientBytecountData> {
+	time_duration: number;
+}
+
+export const isTrusted = (data: any): data is VpnClientTrustedData =>
+	data.common_name != null;
+export const hasBytecountData = (data: any): data is VpnClientBytecountData =>
```

Comment:
It'd be better to have these as
```suggestion
export const hasBytecountData = (data: {}): data is VpnClientBytecountData =>
```
or similar to guarantee that it's an object passed in since the code assumes it'll be an object